src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicy.ts

```diff
@@ -31,6 +31,101 @@ class LinkMediaStats {
+
+  constructor(private videoPriorityBasedPolicyConfig: VideoPriorityBasedPolicyConfig) {
+    this.networkEventChangeTimestamp = 0;
+  }
+
+  getNetworkEventStartTimestamp(
+    numberOfParticipants: number,
+    prevEstimated: number,
+    currEstimated: number
+  ): [number, number] {
+    let networkEventStartTimestamp = 0;
+    let timeBeforeAllowSubscribeMs = 0;
+    const previousNetworkEvent = this.currentNetworkEvent;
+
+    if (currEstimated > prevEstimated) {
+      // if bw increases, we use recovery delay
+      this.currentNetworkEvent = NetworkEvent.Increase;
+      timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
+        this.currentNetworkEvent,
+        numberOfParticipants
+      );
+
+      this.networkIncreaseTimestamp = Date.now();
+      // if bw is increasing, we want to look back to find when it starts to increase (the last moment bw is decreasing).
+      // e.g. [500, 400, 300, 400, 500, *600*], we are at 600. We know the bw is increasing since point 300, which is stored in networkDecreaseTimestamp.
+      networkEventStartTimestamp = this.networkDecreaseTimestamp;
+    } else if (currEstimated < prevEstimated) {
+      // if bw decreases, we use response delay
+      this.currentNetworkEvent = NetworkEvent.Decrease;
+      timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
+        this.currentNetworkEvent,
+        numberOfParticipants
+      );
+
+      this.networkDecreaseTimestamp = Date.now();
+      // if bw is decreasing, we want to look back to find when it starts to decrease (the last moment bw is increasing) and select it as a window.
+      // see example above
+      networkEventStartTimestamp = this.networkIncreaseTimestamp;
+    } else {
+      // if bw is stable, we want to look back to find when it starts to become the current state
+      // e.g. [400, 500, 400, 400, 300, *300*], we are at the last 300. We know the bw is decreasing since point 500, which is stored in networkEventChangeTimestamp
+      networkEventStartTimestamp = this.networkEventChangeTimestamp;
+    }
+
+    if (
+      (previousNetworkEvent &&
+        previousNetworkEvent !== this.currentNetworkEvent) /* network state is changing */ ||
+      this.networkEventChangeTimestamp === 0 /* running the first time */
+    ) {
+      // set networkEventChangeTimestamp to now since network state is changing
+      this.networkEventChangeTimestamp = Date.now();
+    }
+
+    // if networkEventStartTimestamp is not defined, it means the bw is increasing/decreasing ever since the beginning.
+    // e.g. [300, 400, *500*], we are at 500. We set networkEventStartTimestamp to networkDecreaseTimestamp (undefined) above.
+    // So here, we set it to the beginning, which is networkEventChangeTimestamp (explanation above)
+    if (!networkEventStartTimestamp) networkEventStartTimestamp = this.networkEventChangeTimestamp;
```

Comment:
When `networkEventStartTimestamp` is `0` (else case), this condition will also be `true`.

```
if (!0) {
   // Executed!
}
```

If this is intentional, update the comment, e.g., `... is 0 or undefined`. If not, explicitly check for `undefined`.

```
if (typeof networkEventStartTimestamp !== 'undefined') {
```